src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -264,26 +264,25 @@ class ReferralService(
   }
 
   fun getResponsibleProbationPractitioner(referral: Referral): ResponsibleProbationPractitioner {
-    val sentBy = referral.sentBy?.let { it } ?: null
-    return getResponsibleProbationPractitioner(referral.serviceUserCRN, sentBy, referral.createdBy)
+    return getResponsibleProbationPractitioner(referral.serviceUserCRN, referral.sentBy, referral.createdBy)
   }
 
   fun getResponsibleProbationPractitioner(crn: String, sentBy: AuthUser?, createdBy: AuthUser): ResponsibleProbationPractitioner {
     try {
-      val responsibleOfficer = communityAPIOffenderService.getResponsibleOfficer(crn)
-      if (responsibleOfficer.email != null) {
+      val responsibleOfficer = communityAPIOffenderService.getResponsibleOfficerDetails(crn)
+      if (responsibleOfficer?.communityOfficer?.email != null) {
```

Comment:
@marcocolluramoj According to community api team, email is only nullable item and that's why we need to check that

Reply:
Sure, but don't we also need to check whether communityOfficer is the responsible officer since getResponsibleOfficerDetails() is returning the OfficerDetails. Perhaps we could change that to also return null if the communityOfficer isn't a responsible officer by checking the responsibleOfficer flag

Reply:
@marcocolluramoj But we are now throwing InvalidAssumption if there is no responsible officer

Reply:
@kth13 But won't it continue to return a communityOfficer which may not be the responsible officer?